### ISSUE TITLE
ssh into mipsle tmux attach results in corrupting the go runtime

### ISSUE THREAD

**Issue by hdhoang** [tailscale/tailscale#6060]
### What is the issue?

i have a node running tailscale+ssh on openwrt (mipsle). I wanted to start `tmux attach` directly, but client side ssh returns this:

```text
ssh map tmux a
trap:4081, a123456=[1,30851488,0,0,0,0]
results: got {r1=0,r2=4081,errno=0}, want {r1=0,r2=30851488,errno=0
fatal error: AllThreadsSyscall6 results differ between threads; runtime corrupted
trap:4081, a123456=[1,30851488,0,0,0,0]
results: got {r1=0,r2=4081,errno=0}, want {r1=0,r2=30851488,errno=0
fatal error: AllThreadsSyscall6 results differ between threads; runtime corrupted
trap:4081, a123456=[1,30851488,0,0,0,0]
results: got {r1=0,r2=4081,errno=0}, want {r1=0,r2=30851488,errno=0
fatal error: AllThreadsSyscall6 results differ between threads; runtime corrupted
trap:4081, a123456=[1,30851488,0,0,0,0]
results: got {r1=0,r2=4081,errno=0}, want {r1=0,r2=30851488,errno=0
fatal error: AllThreadsSyscall6 results differ between threads; runtime corrupted

goroutine 0 [idle]:
runtime.fatal({0xb8af82, 0x44})
        runtime/panic.go:1066 +0x54
runtime.runPerThreadSyscall()
        runtime/os_linux.go:884 +0x3e0
runtime.sighandler(0x21, 0x1c53d08, 0x1c53d88, 0x1c005a0)
        runtime/signal_unix.go:647 +0x710
runtime.sigtrampgo(0x21, 0x1c53d08, 0x1c53d88)
        runtime/signal_unix.go:479 +0x208
runtime.sigtramp()
        runtime/sys_linux_mipsx.s:325 +0x40

goroutine 1 [running]:
        goroutine running on other thread; stack unavailable

goroutine 0 [idle]:
runtime.fatal({0xb8af82, 0x44})
        runtime/panic.go:1066 +0x54
runtime.runPerThreadSyscall()
        runtime/os_linux.go:884 +0x3e0
runtime.sighandler(0x21, 0x1c5bd08, 0x1c5bd88, 0x1c00780)
        runtime/signal_unix.go:647 +0x710
runtime.sigtrampgo(0x21, 0x1c5bd08, 0x1c5bd88)
        runtime/signal_unix.go:479 +0x208
runtime.sigtramp()
        runtime/sys_linux_mipsx.s:325 +0x40

goroutine 0 [idle]:
runtime.fatal({0xb8af82, 0x44})
        runtime/panic.go:1066 +0x54
runtime.runPerThreadSyscall()
        runtime/os_linux.go:884 +0x3e0
runtime.sighandler(0x21, 0x1c8bd08, 0x1c8bd88, 0x1c820f0)
        runtime/signal_unix.go:647 +0x710
runtime.sigtrampgo(0x21, 0x1c8bd08, 0x1c8bd88)
        runtime/signal_unix.go:479 +0x208
runtime.sigtramp()
        runtime/sys_linux_mipsx.s:325 +0x40

goroutine 0 [idle]:
runtime.fatal({0xb8af82, 0x44})
        runtime/panic.go:1066 +0x54
runtime.runPerThreadSyscall()
        runtime/os_linux.go:884 +0x3e0
runtime.sighandler(0x21, 0x1c43d08, 0x1c43d88, 0x1c002d0)
        runtime/signal_unix.go:647 +0x710
runtime.sigtrampgo(0x21, 0x1c43d08, 0x1c43d88)
        runtime/signal_unix.go:479 +0x208
runtime.sigtramp()
        runtime/sys_linux_mipsx.s:325 +0x40
```

server bugreport:

```
./run/tailscale bugreport --record
The initial bugreport is below; please reproduce your issue and then press Enter...
BUG-d7e2d519c5f34c52a4d48ea0f9f191ae98f0f33c9a4691b9396c30673a400dae-20221025174241Z-4da89c6f7c64b939

BUG-d7e2d519c5f34c52a4d48ea0f9f191ae98f0f33c9a4691b9396c30673a400dae-20221025174253Z-ad2cd2a17622524f
Please provide both bugreport markers above to the support team or GitHub issue.
```

client bugreport:

```
$; tailscale bugreport --record 'attach side'
Recording started; please reproduce your issue and then press Enter...

BUG-7f4f2454cd836548257cd5a87ad14dc809172834a482db9c30c8a77e26fad0fd-20221025174234Z-2115b6ef7a3d8dc0
BUG-7f4f2454cd836548257cd5a87ad14dc809172834a482db9c30c8a77e26fad0fd-20221025174255Z-3ef95bc6b2d11ef8
Please provide both bugreport markers above to the support team or GitHub issue.
```

### Steps to reproduce

ssh to openwrt/mipsle with any terminal-UI command (eg tmux, vi)

### Are there any recent changes that introduced the issue?

plain ssh (without tailscale ssh) worked previously. i think tailscale ssh+TUI app never worked for me. sorry for dragging this until 1.32 mentioned openwrt!

As a workaround, i run ssh to commandline, then do `tmux a` from there.

### OS

Linux, Windows, Android, Other

### OS version

openwrt 22.03 (mipsle), windows ssh, wezterm ssh, fedora ssh, android Juicessh

### Tailscale version

1.32.x

### Bug report

BUG-d7e2d519c5f34c52a4d48ea0f9f191ae98f0f33c9a4691b9396c30673a400dae-20221025174241Z-4da89c6f7c64b939

**Comment by bradfitz** [comment]
Ouch. I filed https://github.com/golang/go/issues/56426 to see if anybody there has any ideas.

**Comment by bradfitz** [comment]
Go fix out for review: https://go-review.googlesource.com/c/go/+/452975

**Comment by bradfitz** [comment]
This should be fixed in the latest Tailscale unstable (dev) builds which are now using Go 1.20 with the fix.

The next stable release with this fix will be Tailscale 1.38.0.
